packages/core/admin/admin/src/content-manager/pages/ListView/components/BulkActionButtons/SelectedEntriesModal/index.js

```diff
@@ -22,23 +23,65 @@ import {
   useFetchClient,
   useQueryParams,
 } from '@strapi/helper-plugin';
-import { Pencil, CrossCircle, CheckCircle } from '@strapi/icons';
+import { Pencil, CrossCircle, CheckCircle, Check } from '@strapi/icons';
 import PropTypes from 'prop-types';
-import { stringify } from 'qs';
 import { useIntl } from 'react-intl';
 import { useQuery } from 'react-query';
 import { useSelector } from 'react-redux';
 import { Link, useHistory } from 'react-router-dom';
 import styled from 'styled-components';
 
+import InjectionZoneList from '../../../../../components/InjectionZoneList';
 import { getTrad, createYupSchema } from '../../../../../utils';
 import { listViewDomain } from '../../../selectors';
 import { Body } from '../../Body';
+import ConfirmBulkActionDialog, { confirmDialogsPropTypes } from '../ConfirmBulkActionDialog';
 
 const TypographyMaxWidth = styled(Typography)`
   max-width: 300px;
 `;
 
+/* -------------------------------------------------------------------------------------------------
+ * ConfirmDialogPublishAll
+ * -----------------------------------------------------------------------------------------------*/
+
+const ConfirmDialogPublishAll = ({ isOpen, onToggleDialog, isConfirmButtonLoading, onConfirm }) => {
+  const { formatMessage } = useIntl();
+
+  return (
+    <ConfirmBulkActionDialog
+      isOpen={isOpen}
+      onToggleDialog={onToggleDialog}
+      dialogBody={
+        <>
+          <Typography id="confirm-description" textAlign="center">
+            {formatMessage({
+              id: getTrad('popUpWarning.bodyMessage.contentType.publish.all'),
+              defaultMessage: 'Are you sure you want to publish these entries?',
+            })}
+          </Typography>
+          <InjectionZoneList area="contentManager.listView.publishModalAdditionalInfos" />
+        </>
+      }
```

Comment:
why is this not just `children`? I think that API style is a lot better than using slot props for something that could grow a lot :)

Reply:
In my opinion that refactor is low priority for now. I will try to get to it if there is time but otherwise we might need to do it after we release the beta. If that is the case I will add a ticket to the backlog to be sure we get it done.